File: 0020-valid-parentheses/solution.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char>st = new Stack<char>();
        int len = s.Length;
        bool ck = true;
        for(int i=0; i<len; i++){
            if(s[i]==')'){
                if(st.Count!=0 && st.Peek()=='(')st.Pop();
                else{
                    ck=false;
                    break;
                }
            }
            else if(st.Count!=0 && s[i]=='}'){
                if(st.Peek()=='{')st.Pop();
                else{
                    ck=false;
                    break;
                }
            }
            else if(st.Count!=0 && s[i]==']'){
                if(st.Peek()=='[')st.Pop();
                else{
                    ck=false;
                    break;
                }
            }
            else st.Push(s[i]);
        }
        if(st.Count!=0)ck = false;
        return ck;
    }
}

File: 0020-valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        int len = s.size();
        stack<char>st;
        for(int i=0;i<len;i++){
            if(s[i]=='('||s[i]=='{'||s[i]=='[')st.push(s[i]);
            else{
                if(st.empty())return false;
                if(s[i]==')'){
                    if(st.top()=='(')st.pop();
                    else break;
                }
                else if(s[i]=='}'){
                    if(st.top()=='{')st.pop();
                    else break;
                }
                else{
                    if(st.top()=='[')st.pop();
                    else break;
                }
            }
        }
        if(st.empty())return true;
        return false;
    }
};
